app/controllers/api/v0/shipments_controller.rb

```diff
@@ -44,14 +44,16 @@ def update
 
       def ready
         authorize! :read, Spree::Shipment
-        unless @shipment.ready?
-          if @shipment.can_ready?
-            @shipment.ready!
-          else
-            render(json: { error: I18n.t(:cannot_ready, scope: "spree.api.shipment") },
-                   status: :unprocessable_entity) && return
-          end
+
+        unless @shipment.ready? || @shipment.can_ready?
+          return render(
+            json: { error: I18n.t(:cannot_ready, scope: "spree.api.shipment") },
+            status: :unprocessable_entity
+          )
         end
+
+        @shipment.ready! unless @shipment.ready?
+
```

Comment:
Thanks Neal and Maikel for detangling this one. I'm trying to process it but it hurts my head.. so I'm looking to see if we cover this with specs. It looks like we only cover two out of three conditions, so would it be possible to 
add the third condition to the specs to prove it works before and after this change?

https://github.com/openfoodfoundation/openfoodnetwork/blob/master/spec/controllers/api/v0/shipments_controller_spec.rb#L96-L111

Reply:
Maikel added a spec for testing if the order is ready already. Is that the 3rd condition that you think needs coverage?

1) Check if a shipment that can be readied is readied.
2) Check if an error occurs if shipment isn't ready and can not be readied.
3) Check if shipment is not readied if already readied.

Is there another one I'm missing?

Reply:
🤦  sorry I missed that, all good!